### PR TITLE
Allow offline and online mode for plugins

### PR DESCRIPTION
**Pull request by ahmet2mir**
Hello Pupuli!

In RabbitMQ we could enable a plugin offline or online.
Some plugins should be enabled offline.

See https://www.rabbitmq.com/plugins.html#ways-to-enable-plugins
man rabbitmq-plugins

### CODE REVIEW

File: lib/puppet/type/rabbitmq_plugin.rb [hunk 1]
```diff
@@ -26,6 +32,15 @@
     newvalues(%r{^\S+$})
   end
 
+  newparam(:mode) do
```
Comment (wyardley) [inline lib/puppet/type/rabbitmq_plugin.rb]: `mode` seems a little potentially ambiguous with file modes, but also, I think an `offline` Boolean would probably be cleaner?

Comment (ahmet2mir) [reply lib/puppet/type/rabbitmq_plugin.rb]: First I used a bool but why offline => true and not online => false
It's not like the flag durable which should be true or false.
Then I used a free param "options" where I just put "--offline".
Finally I found that specify offline or online explicitly is easier. 

If you think that it's more clean and simple for a user, why not :)

**Comment by ahmet2mir** [comment]
Offline or online ? At least 2 years z years ago.
But I found a recent commit in rabbit  https://github.com/rabbitmq/rabbitmq-cli/commit/09466e342ae9196d64116e790fa69a63980d7c66

I was using Rabbit 3.5 and just migrated to 3.7.10.
So since 3.6 it seems that thoses flags had 3 values: offline, online or "best" if nothing specified. 

I will add "best" to other options and Boolean can't work here. The name used in rabbit is "mode".

**Comment by wyardley** [comment]
Is it supported in 3.3.x, though? Otherwise you’ll need to wrap it in version specific logic.

**Comment by wyardley** [comment]
Acceptance tests run against vendor version, which is 3.3.x still on some platforms. Eventually, it may make sense to do a breaking chance and move back to using the official RMQ packages by default (which would also involve handling Erlang deps differently), but so far, no one has created a PR to propose that change.

**Comment by ahmet2mir** [comment]
Arf. Ok I will check it tomorrow (hello from France).

**Comment by wyardley** [comment]
Thanks! Getting the acceptance tests running locally can be super helpful for testing (you can play around with different params / versions too).

I personally think it would be cleaner to decouple the param value and the CLI flags a bit more.

**Comment by wyardley** [comment]
The rubocop and spec tests can also be run locally.

**Comment by ahmet2mir** [comment]
@wyardley added tests, move to provider and check version.

Found from which version thoses flags exists
```bash
docker run --rm -it centos:7 bash
yum install -y https://www.rabbitmq.com/releases/erlang/erlang-19.0.4-1.el7.centos.x86_64.rpm
yum install -y https://www.rabbitmq.com/releases/rabbitmq-server/v3.3.5/rabbitmq-server-3.3.5-1.noarch.rpm
rabbitmq-plugins --help # ko
yum install -y https://www.rabbitmq.com/releases/rabbitmq-server/v3.4.0/rabbitmq-server-3.4.0-1.noarch.rpm
rabbitmq-plugins --help # ok
```

To get the version, rabbit should running. So I added a function to check it.
If it's not running, this option doesn't matter.

**Comment by bastelfreak** [comment]
@wyardley are you fine with this as well?

**Comment by wyardley** [comment]
Seems ok to me from a quick look, just wondering if it makes since to wait til after we cut the release and do modulesync?

**Comment by ahmet2mir** [comment]
@wyardley squashed. Thanks for your time :)

**Comment by wyardley** [comment]
@ahmet2mir Can you rebase again? We've skipped the puppet6-nightly stuff in Travis, at least for now.

**Comment by ahmet2mir** [comment]
@wyardley  done.

**Comment by wyardley** [comment]
Looks like you merged vs rebased?

**Comment by ahmet2mir** [comment]
Arf, I "pull -r" (wrong reflex), fixed with a simple rebase.

**Comment by wyardley** [comment]
That worked! These aren’t the nightly build failure issue, so can you take a look at the Travis beaker failures?

**Comment by ahmet2mir** [comment]
It's something else.
I've got frozen string error. All other tests fails.

```
/home/travis/build/voxpupuli/puppet-rabbitmq/spec/spec_helper_acceptance.rb:6: warning: already initialized constant UNSUPPORTED_PLATFORMS
/home/travis/build/voxpupuli/puppet-rabbitmq/spec/spec_helper_acceptance.rb:6: warning: previous definition of UNSUPPORTED_PLATFORMS was here
An error occurred while loading ./spec/acceptance/clustering_spec.rb.
Failure/Error: run_puppet_install_helper
FrozenError:
  can't modify frozen String
```

**Comment by wyardley** [comment]
@ahmet2mir I'm getting that error on the modulesync PR as well (even though last run on master went Ok), so I think this must be unrelated to changes here.

**Comment by wyardley** [comment]
Thanks for your patience and for the contribution @ahmet2mir

**Pull request merged**